File: rocket_back/rocket/src/main/java/com/uber/rocket/dto/RideDetails.java
package com.uber.rocket.dto;

import lombok.*;

import java.util.List;
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class RideDetails {

    Long id;
    String driver;
    List<String> passengers;
    String driverProfileImage;
    String start;
    String end;
    String price;
    String duration;
    double rating;
    String date;
    List<ReviewDTO> reviews;
}

File: rocket_front/rocket/src/app/interfaces/RideDetails.ts
export interface RideDetails {
  id: string;
  driver: string;
  passengers: string[];
  routeLocation: string;
  driverProfileImage: string;
  start: string;
  end: string;
  price: number;
  duration: number;
  rating: number;

  date: string;
  reviews: {
    reviewer: string;
    rating: number;
    text: string;
    reviewerProfileImage: string;
  }[];
}
